airflow/providers/google/marketing_platform/hooks/display_video.py

```diff
@@ -127,3 +144,77 @@ def run_query(self, query_id: str, params: Dict[str, Any]) -> None:
             .runquery(queryId=query_id, body=params)
             .execute(num_retries=self.num_retries)
         )
+
+    def create_sdf_download_operation(self, body_request: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Creates an SDF Download Task and Returns an Operation.
+        :param body_request: Body request.
+        :type body_request: Dict[str, Any]
+
+        More information about body request n be found here:
+        https://developers.google.com/display-video/api/reference/rest/v1/sdfdownloadtasks/create
+
+        """
+        request = (
+            self.get_conn_to_display_video()  # pylint: disable=no-member
+            .sdfdownloadtasks()
+            .create(body_request=body_request)
+            .execute(num_retries=self.num_retries)
+        )
+        return request["name"]
+
+    def get_sdf_download_operation(self, name: Dict[str, Any]):
+        """
+        Gets the latest state of an asynchronous SDF download task operation.
+        :param name: The name of the operation resource.
+        :type name: str
+        """
+
+        request = (
+            self.get_conn_to_display_video()  # pylint: disable=no-member
+            .sdfdownloadtasks()
+            .operation()
+            .get(name=name)
+            .execute(num_retries=self.num_retries)
+        )
+        return request
+
+    def download_media(self, resource_name: str, filename: str):
```

Comment:
```suggestion
    def _wait_for_media_download(self, resource_name: str, filename: str):
```
Probably should be private

Reply:
I was thinking about it, but this method is used in operator.
I've asked Kamil about it and how to handle it. He told that we do not have specific way to solve such situations. Wdyt?

Reply:
Personally I would do it as I did in `GoogleCampaignManagerDownloadReportOperator`. I don't like the idea of two ctx managers opening the same file. Putting whole download part in execute requires only one context decorator.

Moreover, this download part can be extracted to GoogleBaseHook as it's pretty generic:
```python
def download_content_from_request(file_handle, request, chunk_size):
    downloader = MediaIoBaseDownload(file_handle, request, chunksize=chunk_size)
    done = False
    while done is False:
        _, done = downloader.next_chunk()
    file_handle.flush()
```

Then you just have to do in execute():
```
request = (
    self.get_conn_to_display_video()  # pylint: disable=no-member
    .media()
    .download_media(resource_name=resource_name)
)
with tempfile.NamedTemporaryFile() as temp_file:
    GoogleBaseHook.download_content_from_request(temp_file, request, chunk_size)
    # At this point everything was donwload and is present in temp_file
   upload_to_wherever(temp_file.name)
```